### ISSUE TITLE
Bug(locale): zh-Hant isn't a valid locale format

### ISSUE THREAD

**Issue by wgd3** [ngneat/transloco#636]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Which Transloco package(s) are the source of the bug?

Locale

### Is this a regression?

Yes

### Current behavior

When setting the current language (via TranslocoService.setActiveLang()) to "zh-Hant", TranslocoLocaleService will report "zh-Hant isn't a valid locale format" in the console.

### Expected behavior

"zh-Hant" should be treated as a valid locale so that dates can be formatted properly.

### Please provide a link to a minimal reproduction of the bug, if you won't provide a link the issue won't be handled.

https://stackblitz.com/edit/angular-zyzjqd?file=src/main.ts

### Please provide the environment you discovered this bug in

```markdown
Transloco: 4.1.1
Angular: 14.7.5
Node: 16.18.0
Package Manager: 8.19.2 
OS: macOS 13.2
```

### Additional context

This appears to be a regression from https://github.com/ngneat/transloco/issues/341

In that issue, the PR added support for the "irregular" zh-hant (all lowercase), but not zh-Hant (uppercase H). 

The "irregulars" that are defined are found on this line: https://github.com/ngneat/transloco/blob/master/libs/transloco-locale/src/lib/helpers.ts#L15

### I would like to make a pull request for this bug

Yes 🚀

**Comment by shaharkazaz** [comment]
@wgd3 You are welcome to go ahead an open a PR for it 👍 
Are these locals case sensitive? if not maybe we can modify the regex.

**Comment by wgd3** [comment]
@shaharkazaz happy to open the PR! I found [this RFC spec](https://www.rfc-editor.org/rfc/rfc9110.html#name-language-tags) which leads me to believe that all locales should be case-insensitive. Unless there’s any objection to that I’ll focus on the regex in the PR.

**Comment by wgd3** [comment]
Closed as part of #638

**Comment by shaharkazaz** [comment]
@wgd3 I'll release a version later today.

**Comment by wgd3** [comment]
@shaharkazaz awesome, thank you! Apologies if I wasn't supposed to close this out yet, just a habit from other projects.

**Comment by shaharkazaz** [comment]
Released in v4.0.1
